Add a blog post describing the 1.4/1.5 release timeline and features

aljoscha opened the pull request.

zentol commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+backpressuring for improved checkpoint alignments.
+
+In our testing ([see slide 26 here](https://www.slideshare.net/FlinkForward/flink-forward-berlin-2017-nico-kruber-building-a-network-stack-for-optimal-throughput-lowlatency-tradeoffs)),
+we’ve seen a substantial improvement in latency using event- driven network I/O, and the community
```
remove spacer after "event-"

zentol commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+other words, to the same machine again) if this is possible. The task can then recover from the
+locally-kept state. This makes it possible to avoid reading all state from the distributed file
+system (which is remote over the network). Especially in applications with very large state, not
+having to read many gigabytes over the network and instead from local disk would result in
```
would -> will?

zentol commented on content/news/2017/08/05/release-1.3.2.html:
```diff
@@ -139,7 +139,8 @@
 
       <article>
-        <p>05 Aug 2017</p>
+        <p>05 Aug 2017
```
would be neat if you could revert these; the next regeneration may reverse them again.

HolyHaddock commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+developers has its own set of common tools, deployment models, and expected behavior when working
+with a stream processing framework like Flink.
+
+FLIP-6 will ensure that Flink fits naturally in both of these contexts, behaving although it’s
```
"as though"?

aljoscha commented:
Thanks for the comments, @zentol and @HolyHaddock, I addressed them.

@zentol That change with the `</p>` tags was caused by the multiple-authors change, I fixed that as well.

bowenli86 commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+
+The Apache Flink 1.4.0 release is on track to happen in the next couple of weeks, and for all of the
+readers out there who haven’t been following the release discussion on [Flink’s developer mailing
+list](http://flink.apache.org/community.html#mailing-lists), we’d like to provide some detail on
```
detail**s**

aljoscha replied:
fixing

bowenli86 commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+
+Modern, stateful stream processing has served as a convergence for these two developer communities.
+Despite a significant overlap of the core concepts in the applications being built, each group of
+developers has its own set of common tools, deployment models, and expected behavior when working
```
behavior**s**

bowenli86 commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+developers has its own set of common tools, deployment models, and expected behavior when working
+with a stream processing framework like Flink.
+
+FLIP-6 will ensure that Flink fits naturally in both of these contexts, behaving as though it’s
```
in both contexts and behaves as though....

bowenli86 commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+
+Speed will always be a key consideration for users who build stream processing applications, and
+Flink 1.5 will include a rework of the network stack that will even further improve Flink's latency.
+At the heart of this work is a transition to event-driven network I/O, ensuring that Flink will use
```
can you explain here this is a transition from what to event-driven network I/O?

wints replied:
@aljoscha @bowenli86 proposed change to this section, similar to my comment above. 

"At the heart of this work is a transition from configurable, fixed-interval network I/O to event-driven network I/O and application-level flow control, ensuring that Flink..."

bowenli86 commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,136 @@
+completed and that will have a meaningful impact on Flink’s operability and performance.
+
+1. Rework of the deployment model and distributed process
+2. Transition to event-driven network I/O and application-level flow control for better backpressure handling
```
can you explain here this is a transition from what to event-driven network I/O?

wints replied:
@aljoscha @bowenli86 

A proposed change:

"Transition from configurable, fixed-interval network I/O to event-driven network I/O and application-level flow control for better backpressure handling"

In the status quo, Flink uses a static output flusher that flushes buffers that aren't full every 100ms (this time interval is configurable). In the future, there'd be no more configuration parameter and Flink would get the best possible performance based on resources available at any given time.

aljoscha commented:
I pushed changes to address those comments.

bowenli86 commented:
👍

rmetzger commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,137 @@
+
+This post will describe how the community plans to get there and the rationale behind the approach.
+
+## Coming soon: Major Changes to the Flink’s Runtime
```
"Major changes to Flink's Runtime" (remove the "the")

aljoscha replied:
fixing

rmetzger commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,137 @@
+
+Next, we’ll go through each of these improvements in more detail.
+
+## Reworking Flink’s Deployment Model and Distributed Process
```
"Distributed Processing"?

aljoscha replied:
@wints what do you think?

wints replied:
+1 to changing to "distributed processing"

rmetzger commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,137 @@
+
+## Reworking Flink’s Deployment Model and Distributed Process
+
+[FLIP-6](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=65147077) is an initiative
```
I would explain what "FLIP" stands for.

aljoscha replied:
I think  going to the link explains it. Or we could just call the link "design document".

aljoscha replied:
adding a comment in parentheses

rmetzger commented on _posts/2017-11-21-release-1.4-and-1.5-timeline.md:
```diff
@@ -0,0 +1,137 @@
+network capacity, as well as credit-based flow control which offers more fine-grained backpressuring
+for improved checkpoint alignments.
+
+In our testing ([see slide 26 here](https://www.slideshare.net/FlinkForward/flink-forward-berlin-2017-nico-kruber-building-a-network-stack-for-optimal-throughput-lowlatency-tradeoffs)),
```
Afaik Slideshare supports linking to individual slides immediately, like this:
https://www.slideshare.net/FlinkForward/flink-forward-berlin-2017-nico-kruber-building-a-network-stack-for-optimal-throughput-lowlatency-tradeoffs#26

aljoscha replied:
fixing
